testingii/src/App.js

```diff
@@ -0,0 +1,104 @@
+import React, { Component } from 'react';
+import './App.css';
+import Display from './display/Display.js'
+import Dashboard from './dashboard/Dashboard.js'
+
+class App extends Component {
+  state = {
+    balls: 0,
+    strikes: 0,
+    outs: 0 
+  }
+
+  onStrike = (e) => {
+    let { outs, strikes, balls } = this.state
+    if(outs > 3 || outs < 0 || balls > 4 || balls < 0
+      || strikes > 3 || strikes < 0){
+        console.log('Display has an error')
+    }else if(outs === 2 && strikes > 2){
+      this.setState({
+        balls: 0,
+        strikes: 0,
+        outs: 0 
+      })
+      alert('New Inning');
+    }else if(strikes < 2){
+      this.setState({
+      strikes: strikes+1
+    })
+    }else if(strikes > 2 && outs < 2){
+      this.setState({
+        balls: 0,
+        strikes: 0,
+        outs: outs+1 
+      })
+      alert('Strike out')
+    }
+  }
+
+  onBall = (e) => {
+    e.preventDefault();
+    let { outs, strikes, balls } = this.state
+    if(outs > 3 || outs < 0 || balls > 4 || balls < 0
```

Comment:
I am having difficulty understanding why you are checking for this.  It seems to me like this type of error-handling/validation is redundant. If there is a display error, it would be apparent to the user without you checking for it and console-logging it.  I would remove this if statement from all of your functions. Remember, the goal is to code DRY.